File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/java/io/github/lishangbu/avalon/oauth2/common/core/AuthorizationGrantTypeSupport.java
package io.github.lishangbu.avalon.oauth2.common.core;

import java.io.Serial;
import java.io.Serializable;
import org.springframework.security.oauth2.core.AuthorizationGrantType;

/// 一些过时但仍需支持的授权类型常量
///
/// 提供对密码模式等旧授权类型的兼容支持
///
/// @author lishangbu
public final class AuthorizationGrantTypeSupport implements Serializable {
    @Serial private static final long serialVersionUID = 1L;

    public static final AuthorizationGrantType PASSWORD = new AuthorizationGrantType("password");
    public static final AuthorizationGrantType SMS = new AuthorizationGrantType("sms");
    public static final AuthorizationGrantType EMAIL = new AuthorizationGrantType("email");
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/kotlin/io/github/lishangbu/avalon/oauth2/common/core/AuthorizationGrantTypeSupport.kt
package io.github.lishangbu.avalon.oauth2.common.core

import org.springframework.security.oauth2.core.AuthorizationGrantType
import java.io.Serializable

/**
 * 一些过时但仍需支持的授权类型常量 提供对密码模式等旧授权类型的兼容支持
 *
 * @author lishangbu
 */
object AuthorizationGrantTypeSupport : Serializable {
    /** 序列化版本号 */
    private const val serialVersionUID = 1L

    /** 密码 */
    @JvmField
    val PASSWORD = AuthorizationGrantType("password")

    /** 短信 */
    @JvmField
    val SMS = AuthorizationGrantType("sms")

    /** 邮箱 */
    @JvmField
    val EMAIL = AuthorizationGrantType("email")
}
